internal/webhooks/cluster.go

```diff
@@ -458,7 +459,8 @@ func DefaultVariables(cluster *clusterv1.Cluster, clusterClass *clusterv1.Cluste
 	if clusterClass == nil {
 		return field.ErrorList{field.InternalError(field.NewPath(""), errors.New("ClusterClass can not be nil"))}
 	}
-	defaultedVariables, errs := variables.DefaultClusterVariables(cluster.Spec.Topology.Variables, clusterClass.Spec.Variables,
+
+	defaultedVariables, errs := variables.DefaultClusterVariables(cluster.Spec.Topology.Variables, clusterClassStatusVariablesToVariables(clusterClass.Status.Variables),
```

Comment:
Q: wouldn't it be cleaner to pass in `clusterClass.Status.Variables` without converting them?

Don't we have to do this eventually anyway? (I don't know the current plan about how this code should look eventually)

Reply:
Or all just temporary and will be dropped once we handle From?

Reply:
Yeah - exactly right. I've added todos in the code to signal this more clearly.